Octave 10.3 C++ warnings

With Octave 10.3 locally on Debian unstable (I don't see an easy way to test with this in CI) I see a lot of warnings from the testsuite compiling generated C++ code which look concerning:

```
/usr/include/octave-10.3.0/octave/../octave/ov-base.h:780:30: warning: ‘virtual bool octave_base_value::save_binary(std::ostream&, bool)’ was hidden [-Woverloaded-virtual=]
  780 |   virtual OCTINTERP_API bool save_binary (std::ostream& os, bool save_as_floats);
      |                              ^~~~~~~~~~~
```

SWIG defines this method except the last parameter is `bool&` - it looks like this method signature has changed in Octave's C++ API and used to be `bool&` there too.  I'm not sure which Octave version this changed in.  I'd expect SWIG needs to generate different code for newer Octave, though the tests still seem to pass which surprises me.  Perhaps this `save_binary()` method doesn't get exercised by anything in our test suite?

--- a/Lib/octave/octrun.swg
+++ b/Lib/octave/octrun.swg
@@ -949,7 +949,7 @@ SWIGRUNTIME void swig_acquire_ownership_obj(void *vptr, int own);
       return true;
     }
 
-    virtual bool save_binary (std::ostream& os, bool& save_as_floats) {
+    virtual bool save_binary (std::ostream& os, bool save_as_floats) {
       return true;
     }
 
@@ -1237,7 +1237,7 @@ SWIGRUNTIME void swig_acquire_ownership_obj(void *vptr, int own);
     virtual bool load_ascii (std::istream& is)
       { return ptr->load_ascii(is); }
 
-    virtual bool save_binary (std::ostream& os, bool& save_as_floats)
+    virtual bool save_binary (std::ostream& os, bool save_as_floats)
       { return ptr->save_binary(os, save_as_floats); }
 
     virtual bool load_binary (std::istream& is, bool swap, 
@@ -1368,7 +1368,7 @@ SWIGRUNTIME void swig_acquire_ownership_obj(void *vptr, int own);
       return true;
     }
 
-    virtual bool save_binary (std::ostream& os, bool& save_as_floats) {
+    virtual bool save_binary (std::ostream& os, bool save_as_floats) {
       return true;
     }
 

